web-registry/src/lib/wallet/hooks/useOnAddAddress.tsx

```diff
@@ -0,0 +1,135 @@
+import { MutableRefObject, useCallback, useState } from 'react';
+import {
+  ApolloClient,
+  NormalizedCacheObject,
+  useApolloClient,
+} from '@apollo/client';
+import { useQuery } from '@tanstack/react-query';
+import WalletConnect from '@walletconnect/client';
+import { useSetAtom } from 'jotai';
+
+import {
+  addWalletModalConnectAtom,
+  addWalletModalRemoveAtom,
+  addWalletModalSwitchAtom,
+} from 'lib/atoms/modals.atoms';
+import { getPartyByAddrQuery } from 'lib/queries/react-query/registry-server/graphql/getPartyByAddrQuery/getPartyByAddrQuery';
+
+import { DEFAULT_NAME } from 'pages/ProfileEdit/ProfileEdit.constants';
+import { getDefaultAvatar } from 'pages/ProfileEdit/ProfileEdit.utils';
+
+import { chainInfo } from '../chainInfo/chainInfo';
+import { Wallet } from '../wallet';
+import { getWallet } from '../wallet.utils';
+import { WalletConfig } from '../walletsConfig/walletsConfig.types';
+import { AddAddressParams } from './useAddAddress';
+
+type Props = {
+  wallet: Wallet;
+  walletConfigRef: MutableRefObject<WalletConfig | undefined>;
+  walletConnect?: WalletConnect;
+  accountId?: string;
+  addAddress: (params: AddAddressParams) => Promise<void>;
+};
+
+type OnAddAddressType = () => Promise<void>;
+
+// This hook returns a callback meant to be called by the UI (ie: add address button)
+export const useOnAddAddress = ({
```

Comment:
Not sure about the naming here, `useOn` could make us think that it is an effect reacting to an event while it's just returning a callback.
What about simply calling it `useAddAddressCallback`?

Reply:
yeah I wasn't really fond of it either but I wasn't really sure how to differentiate the lower level hook `useAddAddress` from this one, which also returns a callback, so not sure if `useAddAddressCallback` would be better.
Maybe `useHandleAddAddress` since it's basically an event handler?

Reply:
Sounds good to me.